File: CodeSphere.Domain/Models/Freelancer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CodeSphere.Domain.Models
{
    public class Freelancer : User
    {
        public List<Project> CompletedProjects { get; set; }
        public List<Bid> Bids { get; set; }
        public List<Review> ReviewsReceived { get; set; }
        public int ProfileId { get; set; }
        public Profile Profile { get; set; }

        public Freelancer()
        {
            CompletedProjects = new List<Project>();
            Bids= new List<Bid>();
            ReviewsReceived = new List<Review>();

        }
    }
}

File: code-shpere/src/app/models/freelancer.ts
import { Bid } from "./bid";
import { Profile } from "./profile";
import { Project } from "./project";
import { Review } from "./review";

export interface Freelancer {
    id: number;
    name: string;
    rating: number;
    userName: string;
    role: string;
    passwordHash: string;
    email: string;
    bids: Bid[];
    reviewsReceived: Review[];
    completedProjects: Project[];
    profileId: number;
    profile: Profile;
}
